### PR TITLE
VIM-4316 Model Merge

### PR DESCRIPTION
**Pull request by ghking**
#### Ticket

https://vimean.atlassian.net/browse/VIM-4316

#### Ticket Summary

Merge the models from VIMNetworking into VimeoNetworking.

#### Implementation Summary

Merged the models.

Associated PRs:
https://github.vimeows.com/MobileApps/Vimeo-iOS/pull/973
https://github.vimeows.com/MobileApps/Vimeo-tvOS/pull/73

#### How to Test

N/A

### CODE REVIEW

File: VimeoNetworking/VimeoNetworking/Models/VIMAccount.m [hunk 1]
```diff
@@ -40,7 +37,7 @@ @implementation VIMAccount
 
 - (BOOL)isAuthenticated
 {
-    return [self.accessToken length] > 0 && [[self.tokenType lowercaseString] isEqualToString:BearerKey];
```
Comment (alfiehanssen) [inline VimeoNetworking/VimeoNetworking/Models/VIMAccount.m]: Why we remove constants in favor of the hardcoded strings?

File: VimeoNetworking/VimeoNetworking/Models/VIMAccount.m [hunk 2]
```diff
@@ -57,7 +54,7 @@ - (BOOL)isAuthenticatedWithClientCredentials
 
 - (Class)getClassForObjectKey:(NSString *)key
 {
-    if ([key isEqualToString:UserKey])
```
Comment (alfiehanssen) [inline VimeoNetworking/VimeoNetworking/Models/VIMAccount.m]: Same here

File: VimeoNetworking/VimeoNetworking/Models/VIMAccount.m [hunk 3]
```diff
@@ -79,10 +76,10 @@ - (id)initWithCoder:(NSCoder *)aDecoder
     self = [super init];
     if(self)
     {
-        self.accessToken = [aDecoder decodeObjectOfClass:[NSString class] forKey:NSStringFromSelector(@selector(accessToken))];
-        self.tokenType = [aDecoder decodeObjectOfClass:[NSString class] forKey:NSStringFromSelector(@selector(tokenType))];
-        self.scope = [aDecoder decodeObjectOfClass:[NSString class] forKey:NSStringFromSelector(@selector(scope))];
-        self.userJSON = [aDecoder decodeObjectOfClass:[NSDictionary class] forKey:NSStringFromSelector(@selector(userJSON))];
```
Comment (alfiehanssen) [inline VimeoNetworking/VimeoNetworking/Models/VIMAccount.m]: And similar here, why remove the selector syntax?

File: VimeoNetworking/VimeoNetworking/Models/VIMAccount.m [hunk 4]
```diff
@@ -94,9 +91,9 @@ - (id)initWithCoder:(NSCoder *)aDecoder
 - (void)encodeWithCoder:(NSCoder *)aCoder
 {
     [aCoder encodeObject:self.accessToken forKey:NSStringFromSelector(@selector(accessToken))];
-    [aCoder encodeObject:self.tokenType forKey:NSStringFromSelector(@selector(tokenType))];
-    [aCoder encodeObject:self.scope forKey:NSStringFromSelector(@selector(scope))];
-    [aCoder encodeObject:self.userJSON forKey:NSStringFromSelector(@selector(userJSON))];
+    [aCoder encodeObject:self.tokenType forKey:@"tokenType"];
```
Comment (alfiehanssen) [inline VimeoNetworking/VimeoNetworking/Models/VIMAccount.m]: And here

**Comment by alfiehanssen** [comment]
@nicolelehrer  Do you have a minute to skim this PR? I think you've made model changes in the recent past

File: VimeoNetworking/VimeoNetworking/Models/VIMAccount.m (hunk 4 above)
Comment (ghking) [inline VimeoNetworking/VimeoNetworking/Models/VIMAccount.m]: @alfiehanssen Good catch, mustve taken the wrong version for this file. Fixing..

**Comment by nicolelehrer** [comment]
@ghking took a skim, recent models look good 👍

**Comment by huebnerob** [comment]
Looks great @gavin  👍

**Pull request merged**